testutils/channel.go

```diff
@@ -108,3 +110,22 @@ func IntrospectJSON(ch *tchannel.Channel, opts *tchannel.IntrospectionOptions) s
 
 	return string(marshalled)
 }
+
+func getListener(serveTLS bool) (net.Listener, error) {
+	if serveTLS {
+		return getTLSListener()
+	}
+
+	return net.Listen("tcp", "127.0.0.1:0")
+}
+
+func getTLSListener() (net.Listener, error) {
+	cert, err := tls.X509KeyPair(testcert.TestCert, testcert.TestKey)
+	if err != nil {
+		panic(fmt.Sprintf("testutils: getTLSListener: %v", err))
```

Comment:
I feel that panic is good here since we never expect this to fail. 
Go uses same for httptest https://github.com/golang/go/blob/master/src/net/http/httptest/server.go#L151